File: javascript/ais_sdk/asr_bgm.js
var https = require("https");
var utils = require("./utils");
var signer = require("./signer");
var ais = require("./ais");

module.exports = {
    asr_bgm: function (token, url, callback) {

        var endPoint = utils.getEndPoint(ais.IMAGE_SERVICE);
        // 构建请求信息
        var requestBody = JSON.stringify({"url": url});
        var headers = {
            "Content-Type": "application/json",
            "X-Auth-Token": token,
            "Content-Length": requestBody.length
        };
        var options = utils.getHttpRequestEntityOptions(endPoint, "POST", ais.ASR_BGM, headers, );

        var requset = https.request(options, function (response) {

            // 验证服务调用返回的状态是否成功，如果为200, 为成功, 否则失败。
            if (response.statusCode !== 200) {
                console.log('Http status code is: ' + response.statusCode);
            }

            response.on("data", function (chunk) {
                // 返回中文unicode处理
                var result = JSON.parse(chunk.toString());
                callback(JSON.stringify(result));
            })
        });

        requset.on("error", function (err) {
            console.log(err.message);
        });
        requset.write(requestBody);
        requset.end();
    },

    asr_bgm_aksk: function (_ak, _sk, _url, callback) {

        var endPoint = utils.getEndPoint(ais.IMAGE_SERVICE);
        // 配置ak，sk信息
        var sig = new signer.Signer();
        sig.AppKey = _ak;    //构建ak
        sig.AppSecret = _sk; //构建sk

        // 构建请求信息和请求参数信息
        var requestData = {"url": _url};
        var _headers = {"Content-Type": "application/json"};
        var req = new signer.HttpRequest();
        var options = utils.getHttpRequestEntity(sig, req, endPoint, "POST", ais.ASR_BGM, "", _headers, requestData);

        var requset = https.request(options, function (response) {

            // 验证服务调用返回的状态是否成功，如果为200, 为成功, 否则失败。
            if (response.statusCode !== 200) {
                console.log('Http status code is: ' + response.statusCode);
            }
            response.on("data", function (chunk) {
                callback(chunk.toString());
            })
        });

        requset.on("error", function (err) {
            // 返回中文unicode处理
            var result = JSON.parse(chunk.toString());
            callback(JSON.stringify(result));
        });

        requset.write(req.body);
        requset.end();
    }

};

File: python/ais_sdk/asr_bgm.py
# -*- coding:utf-8 -*-

import sys
import json
import ais_sdk.ais as ais
import ais_sdk.utils as utils
import ais_sdk.signer as signer

#
# access asr, asr_bgm,post data by token
#
def asr_bgm(token, url):
    endpoint = utils.get_endpoint(ais.AisService.IMAGE_SERVICE)
    _url = 'https://%s/v1.0/bgm/recognition' % endpoint

    _data = {
        "url": url,
    }

    resp = utils.request_token(_url, _data, token)
    if sys.version_info.major < 3:
        return resp.decode('unicode-escape').encode('utf-8')
    else:
        return resp.decode('unicode_escape')


#
# access asr, asr_bgm,post data by ak,sk
#
def asr_bgm_aksk(_ak, _sk, url):
    endpoint = utils.get_endpoint(ais.AisService.IMAGE_SERVICE)
    _url = 'https://%s/v1.0/bgm/recognition' % endpoint

    sig = signer.Signer()
    sig.AppKey = _ak
    sig.AppSecret = _sk
    _data = {
        "url": url,
    }

    kreq = signer.HttpRequest()
    kreq.scheme = "https"
    kreq.host = endpoint
    kreq.uri = "/v1.0/bgm/recognition"
    kreq.method = "POST"
    kreq.headers = {"Content-Type": "application/json"}
    kreq.body = json.dumps(_data)

    resp = utils.request_aksk(sig, kreq, _url)
    if sys.version_info.major < 3:
        return resp.decode('unicode-escape').encode('utf-8')
    else:
        return resp.decode('unicode_escape')


